pkg/build/strategies/layered/layered.go

```diff
@@ -63,14 +64,25 @@ func (b *Layered) CreateDockerfile(config *api.Config) error {
 	}
 
 	buffer.WriteString(fmt.Sprintf("FROM %s\n", b.config.BuilderImage))
-	buffer.WriteString(fmt.Sprintf("COPY scripts %s\n", locations[0]))
+	// only COPY scripts dir if required scripts are present, i.e. the dir is not empty;
+	// even if the "scripts" dir exists, the COPY would fail if it was empty
+	_, err = os.Stat(filepath.Join(config.WorkingDir, api.UploadScripts, api.Assemble))
```

Comment:
I had interpreted the "InstallRequied" method name on the installer interface and meaning the script had to be provided if you were providing any scripts.

So a customer can provide only 1 of the required scripts through , and the others say provided by the defaults in STI_SCRIPTS_PATH ?

Yeah, looking back at the code, yes, I now see that it consolidates the scripts from the various potential sources.

OK, truly see if the dir is empty.  Found a src snippet on stackoverflow ... will go that route.